### PR TITLE
Pawn valid move

### PR DESCRIPTION
**Pull request by fistoriza**

### CODE REVIEW

File: app/views/games/show.html.erb [hunk 1]
```diff
@@ -9,18 +9,26 @@
...
                   <div class="piece" data-piece-id="<%= piece.id %>">
-                    <%= piece.name %>
+                    <%= piece.icon.html_safe %>
                   </div>
                 <% end %>
             </td>
           <% else %>
             <td class="white_tile" data-x-pos="<%= col %>" data-y-pos="<%= row %>">
-                <% if piece.blank? == false %>
+<<<<<<< HEAD
```
Comment (ilyakrasnov) [inline app/views/games/show.html.erb]: Please get rid of those lines :) they indicate merge conflicts

**Comment by nnadkarny** [comment]
@fistoriza this is a good start :).  Some thoughts on how to proceed:

- White pawns can only move up (or diagonally up for capture), and black pawns can only move down (or diagonally down for capture).  You'll want to account for that in the `move_vertical?` and `move_diagonal?` methods.

- You can probably remove the horizontal move checks altogether.  Since the only valid moves will be vertical or diagonal-for-capture, any other move besides those two would return false

- You might be able to put the "first_move" logic into its own private method (`pawn_first_move?`), then call that in the `valid_move?` method.

**Comment by nnadkarny** [comment]
One other thought:

- For a pawn to move vertically, there can't be any piece on its target square (including an enemy piece).  You can probably use the is_on_square? method to account for this.

**Comment by fistoriza** [comment]
@nnadkarny all noted. i will update the model. thanks!

**Comment by kirbygit** [comment]
can't do much right now just gonna leave a thumbs up then if you guys think its good just merge it 👍

**Comment by fistoriza** [comment]
Thank you!
Spark by Readdle

**Comment by nnadkarny** [comment]
@fistoriza I'm working on a separate branch to refactor and standardize some of the other piece moves, which should help with the pawn moves as well.  In particular, there will be a `vertical_move_only?` method (used by Rook and Queen) and `vertical_move_one_square?` (used by King).  

I'll be submitting the PR shortly, but if you want to take a look, check out the **refactor-model-methods** branch.  You can find the new standardized move methods in **models/concerns/movements**.

**Comment by nnadkarny** [comment]
@fistoriza I fixed some of the minor CodeClimate issues in the pawn branch.  You can do a git pull to update your local.

Also: the changes I mentioned previously have been merged, you can pull down the latest master to see them.  **models/concerns/movements.rb** has some methods that will be useful for Pawn (particularly `vertical_move_only?` and `vertical_move_one_square?`).  Buzz me on Slack if you have any questions about them.

**Pull request merged**